File: arrayMaxConsecutiveSum/arrayMaxConsecutiveSum.ts
const arrayMaxConsecutiveSum = (inputArray: number[], k: number): number => {

	const sum = (...args: number[]): number => {
		return args.reduce( (a, b) => a + b );
	}

	return inputArray.reduce( (highest, curr, currIdx) => {

		const evalSum: number = sum(
			...inputArray.slice(currIdx, currIdx + k) );

		return evalSum > highest
			? evalSum
			: highest;
	}, 0);
};


const inputs: [] = [
	[[2, 3, 5, 1, 6], 2],
	[[2, 3, 7, 7, 11, 1, 9], 4]
];


console.time('arrayMaxConsecutiveSum x2');

inputs.forEach(inp => {
	console.log(`${inp} -> ${(<any>arrayMaxConsecutiveSum)(...inp)}`);
});

console.timeEnd('arrayMaxConsecutiveSum x2');

File: arrayMaxConsecutiveSum/array_max_consecutive_sum.py

from typing import List


def array_max_consecutive_sum(input_array: List[int], k: int) -> int:
    highest = 0
    for i, n in enumerate(input_array):
        eval_sum = sum(input_array[i:i+k])
        highest = eval_sum if eval_sum > highest else highest

    return highest


inputs = [
    [[2, 3, 5, 1, 6], 2],
    [[2, 3, 7, 7, 11, 1, 9], 4]
]

for inp in inputs:
    print(f'{inp} -> {array_max_consecutive_sum(*inp)}')
